### PR TITLE
feat(lock): near-duplicate lock-dedup hygiene at write time + audit (#1016)

### PR DESCRIPTION
**Pull request by robotrocketscience**
Part of #1016 — the **#1016-C** (lock hygiene) sub-task. Does **not** close the umbrella (frozen/reference tiers, lock-vs-hook docs remain).

## Why

Locks are injected unbounded and never trimmed (#379), so re-locking slightly-reworded ground truth quietly accumulates near-duplicate locks that inflate the injection budget (#1016). Evidence on the real 24-lock store: a **3-lock cluster** that is one fact ("v3.0 #592 eval-harness status") re-locked with κ/kappa wording drift (pairwise J≥0.92, L≥0.88) — 12.5%, matching the ~14% estimate. 2 of those 3 are redundant.

## Design note — dedup, not distillation

The issue's AC-C originally framed hygiene as "char-cap / distillation." Prior R&D found **distillation weak** (locks are already dense, ~1.08× compression) and **dedup/supersession the real lever**, so this implements dedup. Flagging the deviation explicitly.

## What

Reuses the existing `dedup` engine (Jaccard ≥ 0.8 **AND** Levenshtein ≥ 0.85) — no new similarity code:

1. **Write-time guardrail** — `aelf lock` prints a hygiene **warning** when the new lock near-duplicates an existing lock, naming it and suggesting `aelf unlock`/`aelf delete`. Warning only; the lock still writes (it is user-asserted ground truth).
2. **Backlog audit** — `aelf doctor --dedup --dedup-locks` scopes the dedup audit to the user-locked set, so the existing cluster is findable without wading through the full-store report. Live store shows the 3-member cluster.

New `dedup.find_near_duplicate_locks()` + a `locked_only` flag on `dedup_audit()`. The default full-store audit is **byte-identical**. **No belief is auto-deleted** — locks are ground truth, so cleanup stays user-confirmed.

## Tests

`tests/test_dedup.py`: `find_near_duplicate_locks` (finds / excludes self / ignores unlocked / below-threshold / blank), `dedup_audit(locked_only=True)` (scopes to locks, default unchanged), and a `--dedup-locks` CLI test. `tests/test_cli_lock_via_worker.py`: write-time warning fires on a near-dup and stays silent otherwise.

## Verification

- `pytest tests/test_dedup.py tests/test_cli_lock_via_worker.py tests/test_lock_management.py tests/test_lock_contract.py tests/test_slash_commands.py` → 249 passed, 1 skipped.
- `uvx vulture … --min-confidence 80` → clean. `uvx typos` → clean.
- Live: `aelf doctor --dedup --dedup-locks` surfaces the 3-lock cluster.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add lock-scoped near-duplicate detection and auditing to improve lock hygiene without changing existing dedup behavior.

New Features:
- Introduce write-time near-duplicate detection for `aelf lock` that warns when a new lock closely matches existing user-locked beliefs.
- Add a lock-scoped dedup audit mode via `aelf doctor --dedup --dedup-locks` that limits the report to user-locked beliefs.

Enhancements:
- Extend the dedup engine with `find_near_duplicate_locks` and a `locked_only` option on `dedup_audit` while keeping the default full-store audit unchanged.
- Update CLI doctor and lock commands to surface lock-specific dedup information, including a flag to scan only locked beliefs and a summary note in the output.

Documentation:
- Document the new lock-dedup hygiene behavior and lock-scoped audit in the v3 changelog.

Tests:
- Add unit and CLI tests covering lock near-duplicate detection, the locked-only dedup audit mode, and the new `--dedup-locks` flag.

## Summary by CodeRabbit

* **New Features**
  * `aelf lock` now warns when a new lock is very similar to an existing one, helping prevent duplicate locks from piling up.
  * `aelf doctor --dedup-locks` adds a locked-only deduplication scan for checking user-locked beliefs separately.

* **Bug Fixes**
  * Improved duplicate detection so near-matches are reported more clearly, while locked items are not automatically removed.

### CODE REVIEW

File: src/aelfrice/cli.py [hunk 1]
```diff
@@ -4950,19 +4972,23 @@ def _cmd_doctor_dedup(args: argparse.Namespace, out: object) -> int:
         ),
     )
 
+    locked_only = bool(getattr(args, "dedup_locks", False))
```
[the remark this answers is not in the document]
Comment (robotrocketscience) [reply src/aelfrice/cli.py]: Dropped the redundant bool() in 33548e95. Kept getattr — it matches the sibling --dedup-jaccard/--dedup-max-pairs reads just above and guards hand-built Namespaces in direct-call tests.
